code/game/objects/items/devices/radio/radio.dm

```diff
@@ -274,6 +274,8 @@ var/global/list/default_medbay_channels = list(
 	//  Fix for permacell radios, but kinda eh about actually fixing them.
 	if(!M || !message) return 0
 
+	if(speaking.flags & NO_TALK_MSG) return 0

```

Comment:
This is *not* the fix. NO_TALK_MSG is *not* for stopping things being *spoken* over radio, it's for stopping the "X talks into their radio" message.